lib/engine/game/g_18_gb/game.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative 'meta'
+require_relative 'entities'
+require_relative 'map'
+require_relative 'scenarios'
+require_relative 'trainphases'
+
+module Engine
+  module Game
+    module G18GB
+      class Game < Game::Base
+        include_meta(G18GB::Meta)
+        include Entities
+        include Map
+        include Scenarios
+        include TrainPhases
+
+        GAME_END_CHECK = { final_train: :current_or, stock_marketMUST_BID_INCREMENT: :current_or }.freeze
+
+        BANKRUPTCY_ALLOWED = false
+
+        BANK_CASH = 99_999
+
+        CURRENCY_FORMAT_STR = '£%d'
+
+        PRESIDENT_SALES_TO_MARKET = true
+
+        MIN_BID_INCREMENT = 5
+        MUST_BID_INCREMENT_MULTIPLE = true
+        ONLY_HIGHEST_BID_COMMITTED = true
+
+        CAPITALIZATION = :full
+
+        SELL_BUY_ORDER = :sell_buy
+
+        SOLD_OUT_INCREASE = false
+
+        NEXT_SR_PLAYER_ORDER = :first_to_pass
+
+        MUST_SELL_IN_BLOCKS = true
+
+        TRACK_RESTRICTION = :restrictive
+
+        HOME_TOKEN_TIMING = :float
+
+        DISCARDED_TRAINS = :remove
+
+        IMPASSABLE_HEX_COLORS = %i[gray red].freeze
+
+        MARKET_SHARE_LIMIT = 100
+
+        MARKET_TEXT = Base::MARKET_TEXT.merge(
+          unlimited: 'May buy shares from IPO in excess of 60%',
+        )
+
+        MARKET = [
+          %w[50o 55o 60o 65o 70p 75p 80p 90p 100p 115 130 160 180 200 220 240 265 290 320 350e 380e],
+        ].freeze
+
+        STOCKMARKET_COLORS = Base::STOCKMARKET_COLORS.merge(
+          unlimited: :yellow,
+        )
+
+        def init_scenario(optional_rules)
+          num_players = @players.size
+          two_east_west = optional_rules.include?(:two_player_ew)
+          four_alternate = optional_rules.include?(:four_player_alt)
+
+          case num_players
+          when 2
+            SCENARIOS[two_east_west ? '2EW' : '2NS']
+          when 3
+            SCENARIOS['3']
+          when 4
+            SCENARIOS[four_alternate ? '4Alt' : '4Std']
+          when 5
+            SCENARIOS['5']
+          else
+            SCENARIOS['6']
+          end
+        end
+
+        def init_optional_rules(optional_rules)
+          optional_rules = super(optional_rules)
+          @scenario = init_scenario(optional_rules)
+          optional_rules
+        end
+
+        def optional_hexes
+          case @scenario['map']
+          when '2NS'
+            self.class::HEXES_2P_NW
+          when '2EW'
+            self.class::HEXES_2P_EW
+          else
+            self.class::HEXES
+          end
+        end
+
+        def num_trains(train)
+          @scenario['train_counts'][train[:name]]
+        end
+
+        def game_cert_limit
+          @scenario['cert-limit']
+        end
+
+        def game_companies
+          scenario_comps = @scenario['companies']
+          self.class::COMPANIES.select { |comp| scenario_comps.include? comp['sym'] }
```

Comment:
use parens same with line 117

Reply:
Not quite clear what you mean here - I think I can only use parentheses on .select when I check a property of the items themselves (company in this case), and here I'm calling a method on scenario_comps so I'm not clear what it would look like.

Reply:
include?(comp['sym'])